### PR TITLE
Added an option for imgui to ignore its off-screen detection code.

### PR DESCRIPTION
**Pull request by DerpleMQ2**
This helps for ISBoxer users who use mini windows that causeis ImGui to reset window positions.

### CODE REVIEW

File: contrib/imgui/imgui.cpp [hunk 1]
```diff
@@ -7169,7 +7169,8 @@ bool ImGui::Begin(const char* name, bool* p_open, ImGuiWindowFlags flags)
         {
             if (!window->ViewportOwned && viewport_rect.GetWidth() > 0 && viewport_rect.GetHeight() > 0.0f)
             {
-                ClampWindowPos(window, visibility_rect);
+                if (!GetIO().ImGuiIgnoreClampWindow)
```
Comment (brainiac) [inline contrib/imgui/imgui.cpp:7172]: You can use `g.IO` here instead of `GetIO()`

File: contrib/imgui/imgui.h [hunk 2]
```diff
@@ -2305,6 +2305,8 @@ struct ImGuiIO
     ImWchar16   InputQueueSurrogate;                // For AddInputCharacterUTF16()
     ImVector<ImWchar> InputQueueCharacters;         // Queue of _characters_ input (obtained by platform backend). Fill using AddInputCharacter() helper.
 
+    bool        ImGuiIgnoreClampWindow;
```
Comment (brainiac) [inline contrib/imgui/imgui.h:2308]: this _is_ ImGui code, so `IgnoreClampWindow` would be sufficient

File: src/main/ImGuiManager.cpp [hunk 3]
```diff
@@ -1203,6 +1203,18 @@ void ImGuiManager_OverlaySettings()
 	mq::imgui::HelpMarker("When enabled, MacroQuest will take over drawing the EQ Cursor when\n"
 		"it is near to or hovering over an ImGui window.");
 
+	if (ImGui::Checkbox("Ignore Window Clamping", &s_imguiIgnoreClampWindow))
+	{
+		WritePrivateProfileBool("Overlay", "ImGuiIgnoreClampWindow", s_imguiIgnoreClampWindow, mq::internal_paths::MQini);
+
+		auto& io = ImGui::GetIO();
+		if (io.ImGuiIgnoreClampWindow != s_imguiIgnoreClampWindow)
```
Comment (brainiac) [inline src/main/ImGuiManager.cpp:1211]: the condition is unnecessary

Comment (brainiac) [reply src/main/ImGuiManager.cpp:1211]: You also need to set this when the context is created

**Review by brainiac — CHANGES_REQUESTED** [review]
left some comments

File: src/main/ImGuiManager.cpp (hunk 3 above)
Comment (DerpleMQ2) [reply src/main/ImGuiManager.cpp:1211]: fixed.

File: contrib/imgui/imgui.h (hunk 2 above)
Comment (DerpleMQ2) [reply contrib/imgui/imgui.h:2308]: fixed.

File: contrib/imgui/imgui.cpp (hunk 1 above)
Comment (DerpleMQ2) [reply contrib/imgui/imgui.cpp:7172]: fixed.

**Comment by DerpleMQ2** [comment]
Made the changes in the second commit

**Review by brainiac — APPROVED** [review]

**Pull request merged**